cg/store/api/import_func.py

```diff
@@ -106,69 +129,72 @@ def prices_are_same(first_price, second_price):
+def versions_are_same(
+    version_obj: models.ApplicationVersion, application_version: ApplicationVersionSchema
+) -> bool:
     """Checks if the given versions are to be considered equal"""
-
     return (
-        version_obj.application.tag == _get_tag_from_raw_version(raw_version)
-        and version_obj.valid_from
-        == datetime(*xlrd.xldate_as_tuple(float(raw_version["Valid from"]), datemode))
-        and prices_are_same(version_obj.price_standard, raw_version["Standard"])
-        and prices_are_same(version_obj.price_priority, raw_version["Priority"])
-        and prices_are_same(version_obj.price_express, raw_version["Express"])
-        and prices_are_same(version_obj.price_research, raw_version["Research"])
+        version_obj.application.tag == application_version.app_tag
+        and version_obj.valid_from == application_version.valid_from
+        and prices_are_same(version_obj.price_standard, application_version.standard)
+        and prices_are_same(version_obj.price_priority, application_version.priority)
+        and prices_are_same(version_obj.price_express, application_version.express)
+        and prices_are_same(version_obj.price_research, application_version.research)
     )
 
 
-def applications_are_same(application_obj: models.Application, raw_application: dict):
+def applications_are_same(application_obj: models.Application, application: ApplicationSchema):
     """Checks if the given applications are to be considered equal"""
 
-    return application_obj and application_obj.tag == _get_tag_from_raw_application(raw_application)
+    return application_obj and application_obj.tag == application.tag
 
 
-def add_version_from_raw(
-    application_obj, latest_version, raw_version, sign, store: Store, workbook
-):
-    """Adds an application version from a raw application version record"""
+def add_application_version(
+    application_obj: models.Application,
+    latest_version: Optional[models.ApplicationVersion],
+    version: ApplicationVersionSchema,
+    sign: str,
+    store: Store,
+) -> models.ApplicationVersion:
     new_version = store.add_version(
         application=application_obj,
         version=latest_version.version + 1 if latest_version else 1,
-        valid_from=datetime(
-            *xlrd.xldate_as_tuple(float(raw_version["Valid from"]), workbook.datemode)
-        ),
+        valid_from=version.valid_from,
         prices={
-            "standard": raw_version["Standard"],
-            "priority": raw_version["Priority"],
-            "express": raw_version["Express"],
-            "research": raw_version["Research"],
+            "standard": version.standard,
+            "priority": version.priority,
+            "express": version.express,
+            "research": version.research,
         },
         comment="Added by %s" % sign,
     )
     return new_version
 
 
-def add_application_from_raw(raw_application, sign, store: Store):
+def add_application_object(
```

Comment:
I think this code could just be removed if ApplicationSchema validated the values to turn them into boolean values where needed, and then dumped the kwargs into application model!

Reply:
Same like above, I think that should be done in another PR